### ISSUE TITLE
Breakable block equations

### ISSUE THREAD

**Issue by dvdvgt** [typst/typst#1186]
### Description

I would be awesome to have breakable block equations such that long block equations that do not fit in the remainder of the current page get separated at a line break.

As suggested by `@tinger` on discord, an intuitive solution would be to allow

```
#show math.equation.where(block: true): set block(breakable: true)
```

since the `block` parameter suggests that the equation is wrapped in an `block`.

I am not sure what the implications are of doing that automatically. Being able to use `#pagebreak` inside an equation manually would also be a great help. I am looking forward to hearing your opinions on this!

### Use Case

Here is a screenshot of a possible use case:

![image](https://github.com/typst/typst/assets/40773635/11f2bd84-eb10-4ce5-8fa7-c7b4161c00e6)

A quarter page is wasted here because the block equation is not breakable. Of course, I could split the equation into multiple block equations and insert some comments in between, but then I would lose alignment, and sometimes you just cannot get around having a longer equation.

**Comment by PgBiel** [comment]
The show rule there doesn't work because equations don't use `block` for rendering things; math rendering is kind of an entirely different thing in Typst atm. Proposals like #1125, which aim to make math mode more "self-contained", might bring us closer to common behavior between math and markup modes, which would make this feature request much easier to implement.

**Comment by laurmaedje** [comment]
Solved by #4226
